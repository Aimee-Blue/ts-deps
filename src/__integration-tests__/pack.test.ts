import { existsSync, ensureDir } from 'fs-extra';
import { join } from 'path';
import {
  buildAndPack,
  ROOT,
  unarchiveTarGz,
  emptyDirSafe,
  sortedDirectoryContents,
} from './helpers';

jest.setTimeout(120000);

describe('pack', () => {
  it('should work', async () => {
    const { packageLocation, packageName } = await buildAndPack();

    expect(existsSync(packageLocation)).toBe(true);

    const unarchiveDir = join(ROOT, 'integration-test-pkg');

    await emptyDirSafe(unarchiveDir);
    await ensureDir(unarchiveDir);

    await unarchiveTarGz(join(ROOT, packageName), unarchiveDir);

    const allPackageFiles = await sortedDirectoryContents(unarchiveDir);

    expect(allPackageFiles).toMatchSnapshot();
  });
});
