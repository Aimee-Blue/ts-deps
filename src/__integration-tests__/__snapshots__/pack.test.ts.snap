// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pack should work 1`] = `
Array [
  "package/",
  "package/README.md",
  "package/lib/",
  "package/lib/cli/",
  "package/lib/cli/ts-deps.js",
  "package/lib/config/",
  "package/lib/config/babel.config.js",
  "package/lib/config/jest.config.js",
  "package/lib/config/tsconfig.default.json",
  "package/lib/config/tsconfig.js",
  "package/lib/config/tslint.default.json",
  "package/lib/config/types.d.ts",
  "package/lib/defaults.js",
  "package/lib/helpers/",
  "package/lib/helpers/helpers.js",
  "package/lib/helpers/index.js",
  "package/lib/helpers/pullTemplate.js",
  "package/lib/helpers/unarchive.js",
  "package/lib/index.js",
  "package/lib/options.js",
  "package/lib/scripts/",
  "package/lib/scripts/build.js",
  "package/lib/scripts/clean.js",
  "package/lib/scripts/cli.js",
  "package/lib/scripts/combineCoverage.js",
  "package/lib/scripts/index.js",
  "package/lib/scripts/init.js",
  "package/lib/scripts/patch.js",
  "package/lib/scripts/patchPackage.js",
  "package/lib/scripts/patchText.js",
  "package/lib/scripts/patchTsConfig.js",
  "package/lib/scripts/types.d.ts",
  "package/lib/types.d.ts",
  "package/package.json",
  "package/template-max/",
  "package/template-max/to-copy/",
  "package/template-max/to-copy/.yarnrc",
  "package/template-max/to-copy/babel.config.js",
  "package/template-max/to-copy/commitlint.config.js",
  "package/template-max/to-copy/jest.config.integration.js",
  "package/template-max/to-copy/jest.config.js",
  "package/template-max/to-copy/prettier.config.js",
  "package/template-max/to-copy/release.config.js",
  "package/template-max/to-copy/scripts/",
  "package/template-max/to-copy/scripts/build.js",
  "package/template-max/to-copy/scripts/clean.js",
  "package/template-max/to-copy/scripts/combineCoverage.js",
  "package/template-max/to-copy/tslint.json",
  "package/template-max/to-process/",
  "package/template-max/to-process/gitignore",
  "package/template-max/to-process/package.json",
]
`;
